dask/dataframe/__init__.py

```diff
@@ -159,9 +159,13 @@ def inner_func(*args, **kwargs):
             '  python -m pip install "dask[dataframe]" --upgrade  # or python -m pip install'
         )
         raise ImportError(msg) from e
-    warnings.warn(
-        """The current Dask DataFrame implementation is deprecated. 
-In a future release, Dask DataFrame will use new implementation that
+
+    if dask.config.get("dataframe.query-planning-warning") and not dask.config.get(
+        "dataframe.query-planning"
+    ):
```

Comment:
You are here because `_dask_expr_enabled()` returned False
```suggestion
    if dask.config.get("dataframe.query-planning-warning"):
```